src/Workspaces/Core/Portable/Shared/Extensions/ICodeDefinitionFactoryExtensions.cs

```diff
@@ -88,7 +88,7 @@ internal static partial class ICodeDefinitionFactoryExtensions
 
             yield return CodeGenerationSymbolFactory.CreateConstructorSymbol(
                 attributes: default(ImmutableArray<AttributeData>),
-                accessibility: Accessibility.Public,
+                accessibility: containingTypeOpt.IsAbstractClass() ? Accessibility.Protected : Accessibility.Public,
```

Comment:
@CyrusNajmabadi, this won't use Generate Type refactoring.
One of example of the refactoring:
```csharp
class Program
{
    static void Main()
    {
        Base f = new Foo(2);
    }
}

public abstract class Base {
    public int P { get; set; }

    protected Base(int i) { P = i; }
}
```

will generate:
```csharp
internal class Foo : Base
{
    public Foo(Int32 i) : base(i)
    {
    }
}
```

So, I don't think that it has any case which will generate an abstract class (even editor doesn't provide this option)
![generate_type](https://user-images.githubusercontent.com/23630781/28237270-33983c52-6944-11e7-9e49-ac889c2b2c7e.png)